plasma_framework/contracts/src/utils/FailFastReentrancyGuard.sol

```
@@ -0,0 +1,21 @@
+pragma solidity 0.5.11;
+
+/**
+ * @notice Reentrancy guard that fails immediately when a reentrace occurs
+ * @dev Instead of using the one from openzepplin for the following reasons:
+ *      1. Easier to understand
+ *      2. Fail fast when reentracy occurs make testing easier
```

2. Fail fast when reentracy occurs makes testing easier

---

done